flow-report/hooks.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+// TODO(FR-COMPAT): Use context router to handle steps.
+export function useCurrentStep() {
+  const searchParams = new URLSearchParams(location.search);
+  const step = searchParams.get('step');
+  if (step === null) return null;
+
+  const number = Number(step);
+  if (!Number.isFinite(number)) throw new Error('Invalid step param');
```

Comment:
if this throws then nothing is rendered, in unexpected cases for the report we should avoid fatal errors as much as possible.

```suggestion
  if (!Number.isFinite(number)) {
    console.warn(`Invalid step param`);
    return null;
  }
```